snap/hooks/configure

```diff
@@ -0,0 +1,80 @@
+#! /usr/bin/env sh
+
+# Configure hook for rockcraft.
+# This shell script removes LXD images and instances used by previous versions of rockcraft.
+# Errors in the configure hook can be viewed with `snap tasks --last=install` or `snap tasks --last=refresh`.
+
+# separate lxc output by newlines
+# \n can not be last, or it will be stripped by $() - see shellcheck SC3003
+IFS=$(printf '\n\t')
+
+# craft-providers>=1.7.0<1.8.0 uses instances:
+# - base-instance-rockcraft-buildd-base-v00--be83d276b0c767e3ad60
+# - base-instance-rockcraft-buildd-base-v00--aad8a4673a4513574b4b
+# - base-instance-rockcraft-buildd-base-v00--80d84c0cd14bbbf3d100
+OUTDATED_INSTANCES='base-instance-rockcraft-buildd-base-v00--be83d276b0c767e3ad60
```

Comment:
Note: these names are the same for all users.

The full instance names are very long (e.g. `base-instance-rockcraft-buildd-base-v0.0-craft-com.ubuntu.cloud-buildd-core22`), but get truncated and hashed by craft-providers in a deterministic way.